tests/python/pants_test/backend/project_info/tasks/test_export_dep_as_jar.py

```diff
@@ -632,3 +651,124 @@ def test_transitive_targets(self):
         ]
         assert dependency_spec not in enabled_result["source_dependencies_in_classpath"]
         assert dependency_spec in disabled_result["source_dependencies_in_classpath"]
+
+
+class ExportDepAsJarTestWithCodegenTargets(ExportDepAsJarTest):
+
+    # Version of the scala compiler and libraries used for this test.
+    _scala_toolchain_version = "2.10.5"
+
+    def setUp(self):
+        super().setUp()
+
+        #     A
+        #   /  \
+        #  B    F
+        #   \  /
+        #    C
+        #     \
+        #      D (thrift)
+        #       \
+        #        E
```

Comment:
I love this diagram! Thanks a lot.
I got confused for a bit, thinking that `E` depended on `D`. Could you add an extra line like:
```suggestion
        #        E
        #. A depends on B
```

Just one would be enough, to get the flow of the diagram.